src/sentry/api/serializers/models/organization_member/expand/projects.py

```diff
@@ -1,50 +1,73 @@
 from collections import defaultdict
 from typing import Any, Mapping, MutableMapping, Sequence, cast
 
-from sentry.models import OrganizationMember, ProjectTeam, TeamStatus, User
+from sentry.models import OrganizationMember, OrganizationMemberTeam, ProjectTeam, TeamStatus, User
 
 from ..base import OrganizationMemberSerializer
 from ..response import OrganizationMemberWithProjectsResponse
 
 
 class OrganizationMemberWithProjectsSerializer(OrganizationMemberSerializer):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
-        self.project_ids = set(kwargs.pop("project_ids", []))
+        self.projects = {p.id: p for p in kwargs.pop("projects", [])}
+        self.project_ids = set(self.projects.keys())
         super().__init__(*args, **kwargs)
 
     def get_attrs(
         self, item_list: Sequence[OrganizationMember], user: User, **kwargs: Any
     ) -> MutableMapping[OrganizationMember, MutableMapping[str, Any]]:
-        """
-        Note: For this to be efficient, call
-        `.prefetch_related('teams')`
-        on your queryset before using this serializer
-        """
-
         attrs = super().get_attrs(item_list, user)
 
+        # Get all the OrganizationMemberTeam relations so we can map
+        # the member id to the list of team ids
+        #
+        # Note that we're intentionally only working with `team_id`
+        # to avoid having to fetch the team model as well.
+        member_teams = OrganizationMemberTeam.objects.filter(
+            organizationmember_id__in=[om.id for om in item_list],
+            team_id__in=ProjectTeam.objects.filter(
+                project_id__in=self.project_ids,
+                # make sure to filter only for teams that are visible
+                team__status=TeamStatus.VISIBLE,
+            )
+            .values_list("team_id", flat=True)
+            .distinct(),
+        )
```

Comment:
Nit: Could be slightly more efficient by just fetching `member_id, team_id` instead of the whole object.